plugins/woocommerce/src/Utilities/RestApiUtil.php

```diff
@@ -27,4 +27,89 @@ public function get_endpoint_data( $endpoint, $params = array() ) {
 		$json     = wp_json_encode( $server->response_to_data( $response, false ) );
 		return json_decode( $json, true );
 	}
+
+	/**
+	 * Conditionally loads a REST API namespace based on the current route to improve performance.
+	 *
+	 * This function implements lazy loading for WooCommerce REST API namespaces to prevent loading
+	 * all controllers on every request. It checks if the current REST route matches the namespace
+	 * in order for that namespace to be loaded. If the namespace does not match the current rest
+	 * route, a callback will be registered to possibly load the namespace again on `rest_pre_dispatch`;
+	 * this is done to allow the namespace to be loaded on the fly during `rest_do_request()` calls.
+	 *
+	 * @param string   $route_namespace The namespace to check.
+	 * @param callable $callback        The callback to execute if the namespace should be loaded.
+	 *
+	 * @return void
+	 *
+	 * @internal Do not call this function directly. Backward compatibility is not guaranteed.
+	 */
+	public function lazy_load_namespace( string $route_namespace, callable $callback ) {
+		/**
+		 * Filter whether to lazy load the namespace.  When set to false, the namespace will be loaded immediately during initialization.
+		 *
+		 * @param bool   $should_lazy_load_namespace Whether to lazy load the namespace instead of loading immediately.
+		 * @param string $route_namespace            The namespace.
+		 *
+		 * @since 10.3.0
+		 */
+		$should_lazy_load_namespace = apply_filters( 'woocommerce_rest_should_lazy_load_namespace', true, $route_namespace );
+		if ( $should_lazy_load_namespace ) {
+			$this->attach_lazy_loaded_namespace( $route_namespace, $callback );
+		} else {
+			call_user_func( $callback );
+		}
+	}
+
+	/**
+	 * This is the internal function that implements the logic of wc_rest_lazy_load_namespace(). Its interface
```

Comment:
Where is `wc_rest_lazy_load_namespace`? This function does not exist. The PR description mentions it but its not implemented.